source/develop/release-management/features/ovirt-ovn-provider.md

```diff
@@ -0,0 +1,356 @@
+
+The oVirt OVN provider is a proxy between the oVirt engine and
+the OVN North DB. The provider implements the Openstack Rest API
+to allow it to be used by oVirt using the external network provider
+mechanism. REST queries from oVirt are translated into appropriate
+OVN database queries.
+The provider handles the following requests:
+
+*   GET Networks - retrieves a list of all Logical Switches from the OVN north DB Logical_Switch table
+*   GET Network - retrieves a specific Logical Switch from the OVN north DB Logical_Switch table
+*   POST Network - updates a specific Logical Switch from the OVN north DB Logical_Switch table
+*   DELETE Network - deletes a specific Logical Switch from the OVN north DB Logical_Switch table
+
+*   GET Ports - retrieves a list of all Logical Switch Ports from the OVN north DB Logical_Switch_Port table
+*   GET Port - retrieves a specific Logical Switch Port from the OVN north DB Logical_Switch_Port table
+*   POST Port - updates a specific Logical Switch Port from the OVN north DB Logical_Switch_Port table
+*   DELETE Port - deletes a specific Logical Switch Port from the OVN north DB Logical_Switch_Port table
+
+*   GET Subnets - not implemented (no OVN implementation yet)
+*   GET Subnet - not implemented (no OVN implementation yet)
+*   POST Subnet - not implemented (no OVN implementation yet)
+*   DELETE Subnet - not implemented (no OVN implementation yet)
+
+The oVirt OVN provider can be located anywhere in the system. The address of the
+provider is specified when the provider is added to oVirt, so the two can be on
+any two hosts if needed.
+The provider contacts OVN using a web socket, so the two can also be located on
+separate hosts.
+
+### oVirt OVN VIF Driver
+
+The oVirt OVN VIF driver connects the oVirt vNIC to the proper OVS bridge
+and OVN logical network. When an OVN port (added by the provider) is plugged
+into an OVS bridge, OVN Controller will retrieve information about the port
+and the logical network it belongs to from OVN south DB, and create the required
+open flows on the local OVS switch. The OVN controller will also update the
+binding tables on the OVN south DB, causing all other OVN controllers on other
+hosts to update their open flow tables, to allow connections to the newly added
+port (if ports belonging to this logical network are present on this OVS switch).
+The oVirt OVN VIF driver will also take care of providing the data about the
+OVN ports needed by the GetCapabilites and GetStatistics VDSM commands, and
+protect the OVN switches and ports during the SetupNetworks command.
+The oVirt OVN VIF driver must be located on each oVirt host which supports
+OVN. The VIF driver is invoked by VDSM hooks, contacting the OVN databases to
+fetch the required data and updating the hook XML passed libvirt or the
+oVirt engine.
+
+The following vdsm hooks will need to be implemented:
+
+*   before_device_create
+*   before_nic_hotplug
+*   after_vm_start
+*   after_get_caps
```

Comment:
Please note that touching get_caps hook is of a lower priority. on the testing section, add that attempting to setup a network on the nic(s) that is (are) used for northd or tunnel communication, should be blocked by Vdsm.